journeyStar_shacl.ttl

```diff
@@ -145,7 +145,40 @@ js-shacl:JourneyShape a sh:NodeShape ;
     ] ,
     [
         sh:path js:transitThrough ;
-        sh:node js-shacl:LocationShape ;
+        sh:or ( [ sh:datatype xsd:string ]
+                [ sh:node js-shacl:LocationShape ]
+                [ sh:datatype xsd:anyURI ]
+                [ sh:nodeKind sh:IRI ]) ;
+    ] .
+
+### Shape for Class js:Journey
+#########################################
+js-shacl:JourneyShape a sh:NodeShape ;
+    sh:targetClass js:Journey ;
+    sh:property
+    [
+        sh:path js:origin ;
+        sh:minCount 1 ;
+        sh:maxCount 1 ;
+    ] ,
+    [
+        sh:path js:destination ;
+        sh:minCount 1 ;
+        sh:maxCount 1 ;
+    ] ,
+    [
+        sh:path js:startDate ;
+        sh:maxCount 1 ;
+    ] ,
+    [
+        sh:path js:endDate ;
+        sh:maxCount 1 ;
+    ] ,
+    [
+        sh:path js:destination ;
+        sh:disjoint js:origin ;
+        sh:severity sh:Violation ;
+        sh:message "A Journey is specified that the destination and origin are not the same location."@en;
```

Comment:
change the message to: `"A journey cannot have the same origin and destination. For round trips, the entire journey must be defined in stages!"`